Hardware/Hardware/CardReader/StripeCardReader/PaymentIntentParameters+Stripe.swift

```diff
@@ -17,7 +17,15 @@ extension Hardware.PaymentIntentParameters {
 
         let returnValue = StripeTerminal.PaymentIntentParameters(amount: amountForStripe, currency: self.currency)
         returnValue.stripeDescription = self.receiptDescription
-        returnValue.statementDescriptor = self.statementDescription
+
+        /// Stripe allows the credit card statement descriptor to be nil, but not an empty string
+        /// https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntentParameters.html#/c:objc(cs)SCPPaymentIntentParameters(py)statementDescriptor
+        returnValue.statementDescriptor = nil
+        let descriptor = self.statementDescription ?? ""
+        if !descriptor.isEmpty {
+            returnValue.statementDescriptor = descriptor
+        }
+
```

Comment:
I've made a refactor suggestion in #6047

Reply:
I do think we should clean up the unneeded selfs (and will do so here), but I find the flatMap computed property a little more of a mind warp for the next person maintaining this, so if it is ok, I think I will decline?